rules/coding-style/src/Rector/Throw_/AnnotateThrowablesRector.php

```diff
@@ -252,4 +338,55 @@ private function getThrowingStmtPhpDocInfo(Throw_ $throw): PhpDocInfo
 
         return $stmt->getAttribute(AttributeKey::PHP_DOC_INFO);
     }
+
+    private function getUsesFromReflectedFunction(ReflectionFunction $reflectionFunction): array
+    {
+        $functionNode = $this->parseFunction($reflectionFunction);
+        if ($functionNode === false) {
+            return [];
+        }
+
+        $uses = [];
+        foreach ($functionNode->stmts as $stmt) {
+            if (! $stmt instanceof Use_) {
+                continue;
+            }
+
+            $use = $stmt->uses[0];
+            if (! $use instanceof UseUse) {
+                continue;
+            }
+
+            $parts = $use->name->parts;
+            $uses[$parts[count($parts) - 1]] = implode('\\', $parts);
+        }
+
+        return $uses;
+    }
+
+    private function parseFunction(ReflectionFunction $reflectionFunction)
```

Comment:
You should be albe to get parser service in constructor, no need to create it.

Also, never use `static $something`, its hidden fake property.

Reply:
Ok, I will try...

In the mean time I'm refactoring the entire class as now it has a lot of repetitive code and doesn't pass the checks due to cognitive complexity.